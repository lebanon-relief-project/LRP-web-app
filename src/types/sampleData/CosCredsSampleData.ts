export const mockEnvironmentVars = {
  "cloud-object-storage": [
    {
      "instance_name": "MOCK_COS",
      "credentials": {
        "apikey": "mockapikey",
        "cos_hmac_keys": {
          "access_key_id": "mockaccesskeyid",
          "secret_access_key": "mocksecretaccesskey"
        },
        "endpoints": "mockendpoint",
        "iam_apikey_description": "mockapikeydescription",
        "iam_apikey_name": "mockapikeyname",
        "iam_role_crn": "mockiamrole",
        "iam_serviceid_crn": "mockserviceidcrn",
        "resource_instance_id": "mockresourceinstanceid"
      }
    }
  ]
      }

export const mockEnvironmentVarsTrailingSpaces = {
  "cloud-object-storage": [
    {
      "instance_name": "MOCK_COS",
      "credentials": {
        "apikey": "mockapikey ",
        "cos_hmac_keys": {
          "access_key_id": "mockaccesskeyid ",
          "secret_access_key": "mocksecretaccesskey "
        },
        "endpoints": "mockendpoint ",
        "iam_apikey_description": "mockapikeydescription ",
        "iam_apikey_name": "mockapikeyname ",
        "iam_role_crn": "mockiamrole ",
        "iam_serviceid_crn": "mockserviceidcrn ",
        "resource_instance_id": "mockresourceinstanceid "
      }
    }
  ]
      }


export const mockEnvironmentVarsMissing = {
        "cloud-object-storage": [
        ]
            }