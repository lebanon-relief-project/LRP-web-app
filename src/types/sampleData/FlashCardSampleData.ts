import {
  FlashCard,
  FlashCardsCloudantResponse,
  ResultsResponse,
  ResultsResponseCloudantResponse,
} from "../FlashCard";

export const sampleFlashCard: FlashCard = {
  _id: "bauysdbiuyasd",
  title: "sample title",
  body: "sample body",
  cosUri: "test",
  image: "test image",
};

export const sampleFlashCards: FlashCard[] = [sampleFlashCard];

export const sampleFlashCardsResponse = {
  cards: sampleFlashCards,
};

export const sampleFlashCardsCloudantResponse: FlashCardsCloudantResponse = {
  total_rows: 2,
  offset: 0,
  rows: [
    {
      id: "some random id 1",
      key: "some random key 1",
      value: {},
      doc: {
        _id: "some random _id 1",
        title: "some random title 1",
        body: "some random body 1",
        cosUri: "test",
        image: "random image",
      },
    },
    {
      id: "some random id 2",
      key: "some random key 2",
      value: {},
      doc: {
        _id: "some random _id 2",
        title: "some random title 2",
        body: "some random body 2",
        cosUri: "test",
        image: "random image",
      },
    },
  ],
};

export const sampleResultsCloudantResponse: ResultsResponseCloudantResponse = {
  total_rows: 2,
  offset: 0,
  rows: [
    {
      id: "test_id_1",
      key: "test key",
      value: {},
      doc: {
        _id: "test_id_1",
        _rev: "test_rev_1",
        expl_title: "some title 1",
        expl_body: "some body 1",
        image: "some image 1",
        recommendations: [
          {
            title: "some recommendation title 1",
            body: "some recommendation body 1",
          },
        ],
      },
    },
    {
      id: "test_id_2",
      key: "test key",
      value: {},
      doc: {
        _id: "test_id_2",
        _rev: "test_rev_2",
        expl_title: "some title 2",
        expl_body: "some body 2",
        image: "some image 2",
        recommendations: [
          {
            title: "some recommendation title 2",
            body: "some recommendation body 2",
          },
        ],
      },
    },
  ],
};
