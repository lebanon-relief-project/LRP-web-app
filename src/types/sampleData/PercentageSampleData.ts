import {
  PercentageCloudantResponse,
  PercentageResponse,
} from "../ExperiencePercentage";

export const samplePercentage: PercentageResponse = {
  default_percentage: 0.76,
};

export const samplePercentageCloudantResponse: PercentageCloudantResponse = {
  _id: "test_id_1",
  _rev: "rev_id_1",
  default_percentage: 0.76,
};
