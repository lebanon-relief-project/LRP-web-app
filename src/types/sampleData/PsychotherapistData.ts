import {
  PsychotherapistResponse,
  PsychotherapistCloudantResponse,
} from "../Psychotherapist";

export const samplePsychotherapistsResponse: PsychotherapistResponse = {
  psychotherapists: [
    {
      firstName: "string",
      lastName: "string",
      bio: "string",
      remoteSession: true,
      f2fSession: true,
      location: "string",
      therapyServices: {
        theraphyService: true,
      },
      licenseNumber: "string",
      emergencyPhoneNumber: "string",
      consentData: {
        name: "string",
        data: "string",
        dateOfConsent: new Date(2021, 11, 17),
        deletionRequest: true,
        dateOfDeletionRequest: new Date(2022, 11, 17),
      },
      website: "string",
      email: "string",
      patientGroups: {
        patientGroup: true,
      },
      supportedGenders: {
        Male: true,
        Female: true,
      },
      legalPersonality: "string",
      phoneNumber: "string",
      picture: "string",
      therapistGender: "string",
      languages: ["English"],
      internationalPaymentsOnly: true,
      freeService: true,
      bookingApp: "string",
    },
  ],
};

export const samplePsychotherapistCloudantResponse: PsychotherapistCloudantResponse =
  {
    total_rows: 1,
    offset: 0,
    rows: [
      {
        id: "test_id_1",
        key: "test key",
        value: {},
        doc: {
          firstName: "string",
          lastName: "string",
          bio: "string",
          remoteSession: true,
          f2fSession: true,
          location: "string",
          therapyServices: {
            theraphyService: true,
          },
          licenseNumber: "string",
          emergencyPhoneNumber: "string",
          consentData: {
            name: "string",
            data: "string",
            dateOfConsent: new Date(2021, 11, 17),
            deletionRequest: true,
            dateOfDeletionRequest: new Date(2022, 11, 17),
          },
          website: "string",
          email: "string",
          patientGroups: {
            patientGroup: true,
          },
          supportedGenders: {
            Male: true,
            Female: true,
          },
          legalPersonality: "string",
          phoneNumber: "string",
          picture: "string",
          therapistGender: "string",
          languages: ["English"],
          internationalPaymentsOnly: true,
          freeService: true,
          bookingApp: "string",
        },
      },
    ],
  };
