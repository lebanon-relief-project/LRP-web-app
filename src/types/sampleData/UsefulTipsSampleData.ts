import { UsefulTipsCloudantResponse, UsefulTipsResponse } from "../UsefulTips";

export const sampleUsefulTips: UsefulTipsResponse = {
  usefulTips: [
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
  ],
};

export const sampleUsefulTipsCloudantResponse: UsefulTipsCloudantResponse = {
  _id: "test_id_1",
  _rev: "rev_id_1",
  usefulTips: [
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
    {
      title: "Lorem Ipsum",
      body:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
    },
  ],
};
