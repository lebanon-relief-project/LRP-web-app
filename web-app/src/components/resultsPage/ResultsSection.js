import React from "react";
import styled from "styled-components";
import colours from "../../styles/Colours";
import devices from "../../styles/Devices";
import { FlashCardResultContainer } from "./FlashCardResultContainer";

const ResultsSection = (params) => {
  return (
    <StyledSection>
      <Wrapper id={"wrapper"}>
        <StyledHeader>Hereâ€™s some things that can help</StyledHeader>

        <ContentWrapper id={"content"}>
          {params.results ? (
            <FlashCardResultContainer results={params.results} />
          ) : (
            <></>
          )}
        </ContentWrapper>
      </Wrapper>
    </StyledSection>
  );
};

const ContentWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  min-height: 652px;
  width: 960px;
  max-width: 100%;
`;

const Wrapper = styled.div`
  width: 60%;
  min-height: 730px;
  padding-top: 2rem;
  padding-bottom: 2rem;
  margin: 0 16.7%;
  background: inherit;

  @media (max-width: ${devices.ipadpro}) {
    padding-top: 0px;
    padding-bottom: 0px;
    width: 80%;
    margin: 0 10%;
  }

  @media (max-width: ${devices.mobile}) {
    margin: 0;
    padding-top: 0px;
    padding-bottom: 0px;
    width: 100%;
  }
`;

const StyledSection = styled.section`
  background: ${colours.white};
  margin: 0;
  padding: 0;
  width: 100%;
`;

const StyledHeader = styled.h2`
  font-family: Playfair Display;
  font-style: normal;
  font-weight: bold;
  font-size: 30px;
  line-height: 38px;
  margin: 25px 18px;
`;

export default ResultsSection;
