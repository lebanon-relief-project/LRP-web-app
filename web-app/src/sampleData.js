/*
  Temp data for /api/results
*/
export const recommendation_1 = {
  title: "Allow yourself to feel",
  body:
    "As hard as this might sound, the healthiest way to deal with feelings is to notice them, accept them as a reaction to what has happened, and let them be. Emotions are like waves, they subside once they peak, let them do so, or else they will keep trying to resurface.",
};
export const recommendation_2 = {
  title: "Talk about it",
  body:
    "Share your experience, thoughts, and feelings. It's important that you only do so when you feel ready to. This can be a gradual process. Take your time and do it at whatever pace feels comfortable for you.",
};
export const recommendation_3 = {
  title: "Try these excercises",
  body: "Lorem ipsum...",
};

// Flashcard recommendations
export const sampleResultsFeelingGuilty = {
  _id: "9851d5f91b9f896b097acdd32a5d4e80",
  _rev: "",
  expl_title: "Feeling Guilty",
  expl_body: "",
  image: "feeling-guilty.svg",
  recommendations: [recommendation_1, recommendation_2, recommendation_3],
};
export const sampleResultsAchesPains = {
  _id: "9851d5f91b9f896b097acdd32a5dfed8",
  _rev: "",
  expl_title: "Unexplained aches and pains",
  expl_body: "",
  image: "aches-pains.svg",
  recommendations: [recommendation_3, recommendation_1],
};

// Results response
export const sampleResultsResponse = {
  results: [sampleResultsFeelingGuilty, sampleResultsAchesPains],
};
