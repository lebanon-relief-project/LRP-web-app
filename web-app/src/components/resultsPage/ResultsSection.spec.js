import React from "react";
import { render } from "@testing-library/react";
import ResultsSection from "./ResultsSection";
import "jest-styled-components";
import { sampleResultsResponse } from "../../sampleData";

jest.mock("../../util/util", () => ({
  getCardTitleFromSessionStorage: () => "test title"
}));

describe("the ResultsSection component", () => {
  let container;

  it("should match the snapshot", () => {
    ({ container } = render(
      <ResultsSection results={sampleResultsResponse} />
    ));

    expect(container).toMatchSnapshot();
  });
});
