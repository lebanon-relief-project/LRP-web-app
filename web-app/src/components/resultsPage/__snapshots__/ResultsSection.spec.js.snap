// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ResultsSection component should match the snapshot 1`] = `
.c14 {
  box-shadow: 0 0px 8px 0 rgba(0,0,0,0.2);
  -webkit-transition: 0.3s;
  transition: 0.3s;
  Offset: 0px,1px rgba(0,0,0,0.25) border-radius:10px;
  padding: 6px;
  margin: 10px;
}

.c12 {
  margin-left: 10px;
  font-weight: bold;
}

.c13 {
  margin: 10px;
  font-size: 16px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: auto;
  height: 100%;
  z-index: 1;
  position: absolute;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  z-index: 0;
  border-left: 6px solid rgba(255,197,61,0.2);
  position: absolute;
}

.c5 {
  width: 100%;
  max-width: 270px;
  height: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.c11 {
  height: 100px;
  width: 244px;
  margin: 10px;
}

.c8 {
  color: #002766;
  font-weight: bold;
  font-size: 24px;
  min-width: 270px;
  text-align: left;
  padding-left: 24px;
  padding-top: 30px;
  padding-bottom: 30px;
  background: none;
  border: none;
  border-left: solid;
  border-left-color: #FFC53D;
  border-left-width: 6px;
  cursor: pointer;
  outline: inherit;
}

.c9 {
  opacity: 0.3;
  border-left-color: transparent;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 652px;
  width: 960px;
  max-width: 100%;
}

.c1 {
  width: 60%;
  min-height: 730px;
  padding-top: 2rem;
  padding-bottom: 2rem;
  margin: 0 16.7%;
  background: inherit;
}

.c0 {
  background: #ffffff;
  margin: 0;
  padding: 0;
  width: 100%;
}

.c2 {
  font-family: Playfair Display;
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  line-height: 38px;
  margin: 25px 18px;
}

@media (max-width:1025px) {
  .c7 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 0 16px;
    white-space: nowrap;
  }
}

@media (max-width:1025px) {
  .c6 {
    border: none;
    width: 1000px;
    border-top: 4px solid rgba(255,197,61,0.2);
  }
}

@media (max-width:1025px) {
  .c5 {
    max-width: 100%;
    width: auto;
    height: 78px;
    overflow: auto;
  }
}

@media (max-width:1025px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:1025px) {
  .c10 {
    max-width: 100%;
  }
}

@media (max-width:1025px) {
  .c8 {
    border: none;
    border-top: 4px solid #FFC53D;
    max-width: 100%;
    padding: 0;
    min-width: auto;
    margin-right: 30px;
  }
}

@media (max-width:1025px) {
  .c9 {
    opacity: 0.3;
    border-top: 4px solid transparent;
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:1025px) {
  .c1 {
    padding-top: 0px;
    padding-bottom: 0px;
    width: 80%;
    margin: 0 10%;
  }
}

@media (max-width:576px) {
  .c1 {
    margin: 0;
    padding-top: 0px;
    padding-bottom: 0px;
    width: 100%;
  }
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
      id="wrapper"
    >
      <h2
        class="c2"
      >
        Here are some things that can help
      </h2>
      <div
        class="c3"
        id="content"
      >
        <div
          class="c4"
        >
          <h1
            class="c5"
          >
            <div
              class="c6"
            />
            <div
              class="c7"
            >
              <button
                class="c8"
                data-testid="flashcardTitleButton_0"
              >
                test title
              </button>
              <button
                class="c8 c9"
                data-testid="flashcardTitleButton_1"
              >
                test title
              </button>
              <button
                class="c8 c9"
                data-testid="flashcardTitleButton_2"
              >
                test title
              </button>
            </div>
          </h1>
          <div
            class="c10"
          >
            <img
              class="c11"
              src="feeling-guilty.svg"
            />
            <h1
              class="c12"
            >
              Feeling Guilty
            </h1>
            <p
              class="c13"
            />
            <div
              class="c14"
            >
              <div
                id="result card"
              >
                <h4
                  style="margin: 10px; font-size: 20px;"
                >
                  <b>
                    Allow yourself to feel
                  </b>
                </h4>
                <p
                  style="margin: 10px; font-size: 16px;"
                >
                  It can be helpful for you to accept the feelings that surface by allowing yourself time to process the guilt. This could be achieved by accepting that you are not in control of the event; acknowledging  how you reacted at the time of the event; and understanding that you acted to the best of your ability.
                </p>
              </div>
            </div>
            <div
              class="c14"
            >
              <div
                id="result card"
              >
                <h4
                  style="margin: 10px; font-size: 20px;"
                >
                  <b>
                    Talk about it
                  </b>
                </h4>
                <p
                  style="margin: 10px; font-size: 16px;"
                >
                  Share your experience, thoughts, and feelings. It's important that you only do so when you feel ready to. This can be a gradual process. Take your time and do it at whatever pace feels comfortable for you.
                </p>
              </div>
            </div>
            <div
              class="c14"
            >
              <div
                id="result card"
              >
                <h4
                  style="margin: 10px; font-size: 20px;"
                >
                  <b>
                    Try these excercises
                  </b>
                </h4>
                <p
                  style="margin: 10px; font-size: 16px;"
                >
                  Lorem ipsum...
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
